Lib/xml/dom/xmlbuilder.py

```diff
@@ -248,9 +248,9 @@ def _create_opener(self):
     def _guess_media_encoding(self, source):
         info = source.byteStream.info()
         if "Content-Type" in info:
-            for param in info.getplist():
-                if param.startswith("charset="):
-                    return param.split("=", 1)[1].lower()
+            for param in info.get_params([]):
```

Comment:
the `failobj` is not needed since we already have the `Content-Type`. The alternative is to do:

```py
info = source.byteStream.info()
# import email.message
# assert isinstance(info, email.message.Message)
for ctp_name, ctp_value in info.get_params(()):
    if ctp_name == 'charset':
        return ctp_value.lower()
```

Reply:
Good point!